src/ci/docker/host-x86_64/dist-various-1/Dockerfile

```diff
@@ -73,8 +74,8 @@ RUN env \
     CXX=arm-linux-gnueabihf-g++ CXXFLAGS="-march=armv6 -marm -mfpu=vfp" \
     bash musl.sh armhf && \
     env \
-    CC=arm-linux-gnueabihf-gcc CFLAGS="-march=armv7-a" \
-    CXX=arm-linux-gnueabihf-g++ CXXFLAGS="-march=armv7-a" \
+    CC=arm-linux-gnueabihf-gcc CFLAGS="-march=armv7-a -mfpu=vfp3" \
+    CXX=arm-linux-gnueabihf-g++ CXXFLAGS="-march=armv7-a -mfpu=vfp3" \
```

Comment:
Just curious why this change was needed. Did gcc change the default here or something?

Reply:
I'm not exactly sure what changed, but I do see that a lot of GCC tests explicitly specify this combination. Without it, the musl build fails:

```
+ cd musl-1.2.3
+ ./configure --enable-debug --disable-shared --prefix=/musl-armv7hf
checking for C compiler... arm-linux-gnueabihf-gcc
checking whether C compiler works... no; compiler output follows:
cc1: error: '-mfloat-abi=hard': selected architecture lacks an FPU
```

Reply:
This may be a Ubuntu change -- in 20.04, `arm-linux-gnueabihf-gcc -v` reported configuration `--with-arch=armv7-a --with-fpu=vfpv3-d16`, and in 22.04 that's `--with-arch=armv7-a+fp` (w/o separate fpu).

Reply:
In `man gcc` it says under armv7-a, "+fp The VFPv3 floating-point instructions, with 16 double-precision registers.  The extension +vfpv3-d16 can be used as an alias for this extension." So we should probably just match that and it will be consistent with what we had before.